src/modules/Elsa.Scheduling/Tasks/ResumeWorkflowTask.cs

```diff
@@ -1,38 +1,40 @@
 using Elsa.Workflows.Runtime;
+using Elsa.Workflows.Runtime.Exceptions;
 using Elsa.Workflows.Runtime.Messages;
 using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
 
 namespace Elsa.Scheduling.Tasks;
 
 /// <summary>
 /// A task that resumes a workflow.
 /// </summary>
-public class ResumeWorkflowTask : ITask
+public class ResumeWorkflowTask(ScheduleExistingWorkflowInstanceRequest request) : ITask
 {
-    private readonly ScheduleExistingWorkflowInstanceRequest _request;
-
-    /// <summary>
-    /// Initializes a new instance of the <see cref="ResumeWorkflowTask"/> class.
-    /// </summary>
-    /// <param name="request">The dispatch workflow instance request.</param>
-    public ResumeWorkflowTask(ScheduleExistingWorkflowInstanceRequest request)
-    {
-        _request = request;
-    }
-
     /// <inheritdoc />
     public async ValueTask ExecuteAsync(TaskExecutionContext context)
     {
         var cancellationToken = context.CancellationToken;
         var workflowRuntime = context.ServiceProvider.GetRequiredService<IWorkflowRuntime>();
-        var workflowClient = await workflowRuntime.CreateClientAsync(_request.WorkflowInstanceId, cancellationToken);
-        var request = new RunWorkflowInstanceRequest
+        var logger = context.ServiceProvider.GetRequiredService<ILogger<ResumeWorkflowTask>>();
+
+        try
+        {
+            var workflowClient = await workflowRuntime.CreateClientAsync(request.WorkflowInstanceId, cancellationToken);
+            var runWorkflowInstanceRequest = new RunWorkflowInstanceRequest
+            {
+                Input = request.Input,
+                Properties = request.Properties,
+                ActivityHandle = request.ActivityHandle,
+                BookmarkId = request.BookmarkId
+            };
+            await workflowClient.RunInstanceAsync(runWorkflowInstanceRequest, cancellationToken);
+        }
+        catch (WorkflowInstanceNotFoundException ex)
```

Comment:
i assume there is not a way to find out if the workflow still exists without a exception.

Reply:
Technically, we could add a method to check to see if it exists, but that would then always incur 2 DB calls instead of 1. Given that this usually doesn't happen, unless workflow instances have been deleted and background triggers are about to fire, it's better to catch the exception. Alternatively, we could consider adding a method that returns a result object rather than throwing an exception.